BreakinIn/Messages/ReptIn.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BreakinIn.Messages
+{
+    public class ReptIn : AbstractMessage
+    {
+        public override string _Name { get => "rept"; }
+
+        public string PERS { get; set; }
+        public string PROD { get; set; }
+        public string LANG { get; set; }
+
+        public override void Process(AbstractEAServer context, EAClient client)
+        {
+            var mc = context as MatchmakerServer;
+            if (mc == null) return;
+
+            var PlayerReportingPlayer = new ReptOut()
+            {
+                PERS = PERS,
+            };
+
+            var user = client.User;
+            if (user == null) return;
+
+
+
+            if (user.PersonaName == PERS)
+            {
+                Console.WriteLine(user.PersonaName + " attempted to report themselves, which is not allowed.");
+                client.SendMessage(new ReportingSelf());
+            }
+            else
+            {
+                if (user.CurrentRoom.Name.ToString() == null)
```

Comment:
This case is impossible - When a room is not null, its name will always be defined. There is a chance, though, that the user will not be in a room, in which case `user.CurrentRoom` will be null.

Additionally, calling ToString on a string is redundant, and `((String)null).ToString()` will throw an exception.